test/core/channel/channel_stack_builder_test.cc

```diff
@@ -57,70 +53,57 @@ void CallDestroyFunc(grpc_call_element* /*elem*/,
                      const grpc_call_final_info* /*final_info*/,
                      grpc_closure* /*ignored*/) {}
 
-bool g_replacement_fn_called = false;
-bool g_original_fn_called = false;
-
-void SetReplacementFnCalled(grpc_channel_stack*, grpc_channel_element*) {
-  g_replacement_fn_called = true;
-}
-
-void SetOriginalFnCalled(grpc_channel_stack*, grpc_channel_element*) {
-  g_original_fn_called = true;
+const grpc_channel_filter* FilterNamed(const char* name) {
+  static auto* filters =
```

Comment:
Do we actually need to store the instances in a global like this?  It seems like we could instead just return a `const grpc_channel_filter` by value and store it in a local variable in the individual tests.

Reply:
It simplifies things a bunch in the tests - which I tend to lean towards.